alacritty/src/wayland_theme.rs

```diff
@@ -32,51 +34,61 @@ impl AlacrittyWaylandTheme {
         }
     }
 
-    fn color_icon_color(&self, color: Rgb, status: ButtonState) -> [u8; 4] {
-        match status {
-            ButtonState::Hovered => [0xff, color.r, color.g, color.b],
-            ButtonState::Idle => [0xff, self.foreground.r, self.foreground.g, self.foreground.b],
-            ButtonState::Disabled => {
-                [0xff, self.dim_foreground.r, self.dim_foreground.g, self.dim_foreground.b]
-            },
-        }
-    }
-}
-
-impl WaylandTheme for AlacrittyWaylandTheme {
-    fn primary_color(&self, _window_active: bool) -> [u8; 4] {
-        [0xff, self.background.r, self.background.g, self.background.b]
-    }
-
-    fn secondary_color(&self, window_active: bool) -> [u8; 4] {
+    fn color_icon_color(&self, color: Rgb, status: ButtonState, window_active: bool) -> Rgb {
         if window_active {
-            [0xff, self.foreground.r, self.foreground.g, self.foreground.b]
+            match status {
+                ButtonState::Hovered => color,
+                ButtonState::Idle => self.foreground,
+                ButtonState::Disabled => self.dim_foreground,
+            }
         } else {
-            [0xff, self.dim_foreground.r, self.dim_foreground.g, self.dim_foreground.b]
+            self.dim_foreground
         }
     }
+}
 
-    fn close_button_color(&self, _status: ButtonState) -> [u8; 4] {
-        [0x00, self.background.r, self.background.g, self.background.b]
-    }
-
-    fn close_button_icon_color(&self, status: ButtonState) -> [u8; 4] {
-        self.color_icon_color(self.hovered_close_icon, status)
-    }
+impl WaylandTheme for AlacrittyWaylandTheme {
+    fn element_color(&self, element: Element, window_active: bool) -> ARGBColor {
+        let Rgb { r, g, b } = match element {
+            Element::Bar => self.background,
+            Element::Separator => self.background,
+            Element::Text => {
+                if window_active {
+                    self.foreground
+                } else {
+                    self.dim_foreground
+                }
+            },
+        };
 
-    fn maximize_button_color(&self, _status: ButtonState) -> [u8; 4] {
-        [0x00, self.background.r, self.background.g, self.background.b]
+        ARGBColor { a: 0xff, r, g, b }
```

Comment:
Couldn't we implement a conversion method?  Either just always using alpha of `0xff`, or having an `into_glutin_argb` that takes alpha as a parameter.

I'd recommend putting the method here versus where RGB is defined, since this is the only place where ARGBColor will ever be used I think.

Reply:
the thing is that we use alpha `0x00` sometimes.

Reply:
In that case there's no need to convert it? Because if the alpha is 0, it doesn't matter what you convert it from.